### PR TITLE
Added tests, and fixed a lexical parser edge case

### PR DESCRIPTION
**Pull request by pandu-k**
Edge case was for this:

`"something" "goo` -> required=`["something"]`, optional=`"goo`

But because it is a syntax error, we need: 

`"something" "goo` -> required=`[]`, optional=`"something" "goo`

### CODE REVIEW

File: src/marqo/tensor_search/utils.py [hunk 1]
```diff
@@ -228,8 +241,11 @@ def parse_lexical_query(text: str):
             # None of the above: Syntax error. Interpret text literally instead.
             else:
                 return([], text)
-                
-            
+
+    if opening_quote_idx is not None:
```
Comment (vicilliar) [inline src/marqo/tensor_search/utils.py:245]: great catch on this one @pandu-k

File: tests/tensor_search/test_utils.py [hunk 2]
```diff
@@ -162,3 +162,56 @@ def run():
...
+        # 2-tuples of input text, and expected parse_lexical_query() output
+        cases = [
+            ('just a string', ([], 'just a string')),
+            ('just a "string"', (["string"], 'just a')),
+            ('just "a" string', (["a"], 'just string')),
+            ('"just" a string', (["just"], 'a string')),
+
+            ('朋友你好', ([], '朋友你好')),
+            ('朋友 "你好"', (["你好"], '朋友')),
+            # spaces get introduced, even though Chinese doesn't use them:
```
Comment (vicilliar) [inline tests/tensor_search/test_utils.py:176]: super interesting note

**Review by vicilliar — APPROVED** [review]
